playground/backend/internal/code_processing/code_processing.go

```diff
@@ -426,16 +428,14 @@ func processError(ctx context.Context, errorChannel chan error, pipelineId uuid.
 	return utils.SetToCache(ctx, cacheService, pipelineId, cache.Status, newStatus)
 }
 
-// processCompileError processes error received during processing compile step.
-// This method sets error output and corresponding status to the cache.
-func processCompileError(ctx context.Context, errorChannel chan error, errorOutput []byte, pipelineId uuid.UUID, cacheService cache.Cache) error {
-	err := <-errorChannel
-	logger.Errorf("%s: Compile(): err: %s, output: %s\n", pipelineId, err.Error(), errorOutput)
+// processErrorWithSavingOutput processes error with saving to cache received error output.
+func processErrorWithSavingOutput(ctx context.Context, err error, errorOutput []byte, pipelineId uuid.UUID, subKey cache.SubKey, cacheService cache.Cache, errorTitle string, newStatus pb.Status) error {
+	logger.Errorf("%s: %s(): err: %s, output: %s\n", pipelineId, errorTitle, err.Error(), errorOutput)
 
-	if err := utils.SetToCache(ctx, cacheService, pipelineId, cache.CompileOutput, fmt.Sprintf("error: %s, output: %s", err.Error(), string(errorOutput))); err != nil {
+	if err := utils.SetToCache(ctx, cacheService, pipelineId, subKey, fmt.Sprintf("error: %s, output: %s", err.Error(), string(errorOutput))); err != nil {
```

Comment:
Why do we take `errorOutput` as bytes and then cast it to the string? Is it a way to get the right representation of `errorOutput`?

Reply:
Changed.

Reply:
I see that the `processRunError` below also receives bytes, let's then keep the consistency and leave bytes here as well. What do you think?

Reply:
Changed.